core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1923,4 +1903,116 @@ public void unregisterReceiver(BroadcastReceiver receiver) {
             PlaybackService.this.unregisterReceiver(receiver);
         }
     };
+
+    private boolean isStarted() {
+        return serviceManager.serviceInStartedState;
+    }
+
+    /**
+     * The helper that manages PlaybackService's foreground service life cycle and the associated
+     * notification control.
+     *
+     * The logic is adapted from a sample app from The Android Open Source Project.
+     * See https://github.com/googlesamples/android-MediaBrowserService/blob/6cf01be9ef82ca2dd653f03e2a4af0b075cc0021/Application/src/main/java/com/example/android/mediasession/service/MusicService.java#L211
+     *
+     */
+    private class ServiceManager {
```

Comment:
Might be better to move this to a new class. PlaybackService already is waaay too long.

Reply:
I do agree that PlaybackService is way too long and complex.

I am of two minds of moving `ServiceManager` out of `PlaybackService` as it is  intrinsically tied to the service's internal state. If we do, 
- `ServiceManager` still needs to reference `PlaybackService`
- we might also need to move `setupNotification()` , which accesses the service internal states.